tensorboard/util/string_util.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""TensorBoard helper routines for strings."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import string
+
+
+def sanitize_for_filename(s):
+    """Returns a sanitized filename-friendly version of the input string.
+
+    This discards all characters that are not ASCII letters, digits, or
+    punctuation, and then converts any remaining characters that are not in
+    [-_=,A-Za-z0-9] to "_".
```

Comment:
Doc comment includes comma as allowed character, but code doesn’t?

Reply:
Great catch, thanks.  Updated the code to match docs.